### DISCUSSION TITLE
ShapeShifter Example

### DISCUSSION THREAD

**Discussion by canadyre** [Trusted-AI/adversarial-robustness-toolbox#1375] [Q&A]
**Is your feature request related to a problem? Please describe.**
I have been trying for a while to get the ShapeShifter attack implemented on a custom dataset using the TensorFlowFasterRCNN model as input. It seems that some of the code was written using TF 1.x or something because there are a lot of errors related to this like 'AttributeError: module 'tensorflow' has no attribute 'placeholder''. I would really like to use this toolkit for implementing this attack but it has not been very straightforward.

**Describe the solution you'd like**
Either updates to the code so that tensorflow version matches or give examples of the ShapeShifter attack in a Jupyter Notebook so that we may see a typical workflow for an object detection attack.

**Describe alternatives you've considered**

**Additional context**

**Comment by beat-buesser** [comment]
Hi @canadyre Thank you for your interest in ART!

Yes, `ShapeShifter` and  `TensorFlowFasterRCNN` require TensorFlow 1.15.

I agree, a notebook on `ShapeShifter` on `TensorFlowFasterRCNN` would be great to have, but it may take a while. To get started you could take a look at the tests for `ShapeShifter` using `TensorFlowFasterRCNN`: https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/399640cb26202e2c72056aeb0efd6b93dd5d59da/tests/attacks/test_shapeshifter.py#L57

This test runs in an environment defined by this Dockerfile: https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/.github/actions/tf-faster-rcnn/Dockerfile

Let us know if it works for you and if your code can be wrapped into a notebook, we would be happy to merge it into ART's collection.

**Comment by canadyre** [comment]
Hi @beat-buesser ,

Thank you so much for the response, this looks like it will help a lot. Also, thank you for the clarification on the tensorflow versions. I will update this issue with whether or not I could get it working. 

Thank you
